explorer/chartscache_test.go

```diff
@@ -118,3 +120,103 @@ func TestChartsCache(t *testing.T) {
 		}
 	})
 }
+
+func TestTimeArrSourceSlicingIndex(t *testing.T) {
+	type testData struct {
+		TimeArr      []dbtypes.TimeDef
+		AncestorTime int64
+		index        int
+	}
+
+	var t1, t2, t3, t4, t5, t6 dbtypes.TimeDef
+	t1.T = time.Unix(1548890641, 0)
+	t2.T = time.Unix(1548932462, 0)
+	t3.T = time.Unix(1548977687, 0)
+	t4.T = time.Unix(1549014010, 0)
+	t5.T = time.Unix(1549054190, 0)
+	t6.T = time.Unix(1549096518, 0)
+
+	td := []testData{
+		{
+			TimeArr:      []dbtypes.TimeDef{},
+			AncestorTime: 1548932462,
+			index:        0,
+		},
+		{
+			TimeArr:      []dbtypes.TimeDef{t1, t2, t3, t4, t5, t6},
+			AncestorTime: 0,
+			index:        0,
+		},
+		{
+			TimeArr:      []dbtypes.TimeDef{t1, t2, t3, t4, t5, t6},
+			AncestorTime: 1548890641,
+			index:        0,
+		},
+		{
+			TimeArr:      []dbtypes.TimeDef{t1, t2, t3, t4, t5, t6},
+			AncestorTime: 1548932462,
+			index:        1,
+		},
+		{
+			TimeArr:      []dbtypes.TimeDef{t1, t2, t3, t4, t5, t6},
+			AncestorTime: 1549096518,
+			index:        5,
+		},
+	}
+
+	for i, val := range td {
+		t.Run("Test_#"+strconv.Itoa(i), func(t *testing.T) {
+			resp := timeArrSourceSlicingIndex(val.TimeArr, val.AncestorTime)
+			if val.index != resp {
+				t.Fatalf("expected the returned index to be %d but found it to be %d",
+					val.index, resp)
+			}
+		})
+	}
+}
+
+func TestHeightArrSourceSlicingIndex(t *testing.T) {
+	type testData struct {
+		HeightsArr     []uint64
+		AncestorHeight uint64
+		index          int
+	}
+
+	td := []testData{
+		{
+			HeightsArr:     []uint64{},
+			AncestorHeight: 275156,
+			index:          0,
+		},
+		{
+			HeightsArr:     []uint64{279156, 279157, 279158, 279159, 279160, 279161},
+			AncestorHeight: 0,
+			index:          0,
+		},
+		{
+			HeightsArr:     []uint64{279156, 279157, 279158, 279159, 279160, 279161},
+			AncestorHeight: 279158,
+			index:          2,
+		},
+		{
+			HeightsArr:     []uint64{279156, 279157, 279158, 279159, 279160, 279161},
+			AncestorHeight: 279157,
+			index:          1,
+		},
+		{
+			HeightsArr:     []uint64{279156, 279157, 279158, 279159, 279160, 279161},
+			AncestorHeight: 279161,
```

Comment:
Thanks for adding these!  Please also add a test for when `AncestorHeight` is greater than all of the elements in `HeightsArr`.